src/js/imageWithPreview.directive.js

```diff
@@ -122,6 +123,44 @@
             });
             return deferred.promise;
           };
+          ngModel.$asyncValidators.size = function(modelValue, viewValue) {
+            if (!attrs.size) {
+              return createResolvedPromise();
+            }
+
+            var value = modelValue || viewValue;
+
+            if (!value || !value.fileReaderPromise) {
+              return createResolvedPromise();
+            }
+
+            var deferred = $q.defer();
+
+            value.fileReaderPromise.then(function(dataUrl) {
+
+              var image = document.createElement('img');
```

Comment:
An image element is created both here and in the `dimensions` validator. Depending on the size of images used and frequency with which these validators will get called this may put unnecessary strain on the user's browser. May be a somewhat premature optimization, but this could be solved in a couple of ways.

1. Load the image once and cache the result so that whichever validator executes last can use the cached value.
2. Combine the 2 validations into a single HTML attribute.

Number 2 is easier to implement and less error prone and _may_ end in a better interface. I wouldn't really know about the best API, because I haven't been using this library or Angular in general for a long time.

Just thoughts. Not asking you to change anything.

Reply:
I very much agree. Some of our testing with larger images did result in a slow down here, and even while writing I thought that it is very strange to be re-creating the image object twice.

However I too am not great with AngularJS, and I'm unsure of the best placement for the image creation so that the various validators can access it. This PR was nearly entirely just the previously mentioned PRs work, except with tests that passed.